onnxruntime/core/providers/openvino/openvino_execution_provider.cc

```diff
@@ -29,57 +29,60 @@ OpenVINOExecutionProvider::OpenVINOExecutionProvider(const OpenVINOExecutionProv
   }
   //to check if target device is available
   //using ie_core capability GetAvailableDevices to fetch list of devices plugged in
-  bool device_found = false;
-  bool device_id_found = false;
-  auto available_devices = openvino_ep::BackendManager::GetGlobalContext().ie_core.GetAvailableDevices();
-  //Checking for device_type configuration
-  if (info.device_type_ != "") {
-    if(info.device_type_== "CPU" || info.device_type_== "GPU" || info.device_type_== "MYRIAD") {
-      for (auto device : available_devices){
-        if (device.rfind(info.device_type_, 0) == 0){
-          if(info.device_type_== "GPU" && (info.precision_ == "FP32" || info.precision_ == "FP16")) {
+  if (info.use_compiled_network_ == false) {
```

Comment:
so what we really are doing here is:
1.if the user runs the application with use_compiled_network_=true, we are not doing the device_type check.
The device_type variable gets assigned to device value from global context.

But, what if the machine doesn't have the device for which the user is running the application for. (Even if its with cahced network). It defeats the purpose of our validation device_type check?

<I don't think this is the right change>

2.If we are doing this change for device_type, why are we not doing for device_id check?
Also, I dont think so there's going to be any significant gain with this change for FIL. Just some ~5 to ~10  millisecond gain is my guess.